libs/hdf-converters/src/msft-secure-score-mapper.ts

```diff
@@ -0,0 +1,203 @@
+    ILookupPath
+  > = {
+    platform: {
+      name: 'Heimdall Tools',
+      release: HeimdallToolsVersion
+    },
+    version: HeimdallToolsVersion,
+    statistics: {},
+    profiles: [
+      {
+        name: 'Microsoft Secure Score Scan',
+        title: {
+          transformer: (data: SecureScore) =>
+            `Azure Secure Score report - TenantID: ${data.azureTenantId}`
+        },
+        supports: [],
+        attributes: [],
+        groups: [],
+        status: 'loaded',
+        controls: [
+          {
+            path: 'controlScores',
+            id: {
+              transformer: (data: ControlScore) =>
+                `${data.controlCategory}:${data.controlName}`
+            },
+            title: {
+              transformer: (data: ControlScore) => {
+                const titles = this.getProfiles(data.controlName || '')
+                  .filter((profile) => profile.title !== undefined)
+                  .map((profile) => profile.title);
+
+                if (titles.length > 0) {
+                  return titles.join('... ');
+                } else {
+                  return `${data.controlCategory || ''}:${data.controlName || ''}`;
+                }
+              }
+            },
+            desc: {path: 'description'},
+            impact: {
+              transformer: (data: ControlScore) => {
+                // return controlCategory from the profile document where its id matches the controlName
+                const knownMaxScores = this.getProfiles(
+                  data.controlName || ''
+                ).map((profile) => profile.maxScore || 0);
+
+                if (knownMaxScores.length === 0) {
+                  return 0.5;
+                }
+
+                const highMaxScore = Math.max(...knownMaxScores);
+                return highMaxScore / 10.0;
+              }
+            },
+            refs: [],
+            tags: {
+              group: {
+                transformer: (data: ControlScore) => {
+                  // return controlCategory from the profile document where its id matches the controlName
+                  return this.getProfiles(data.controlName || '').map(
+                    (profile) => profile.controlCategory
+                  );
+                }
+              },
+              tiers: {
+                transformer: (data: ControlScore) => {
+                  // return tiers from the profile document where its id matches the controlName
+                  return this.getProfiles(data.controlName || '').map(
+                    (profile) => profile.tier
+                  );
+                }
+              },
+              threats: {
+                transformer: (data: ControlScore) => {
+                  // return unique threats from the profile document where its id matches the controlName
+                  const uniqs: Set<string> = new Set();
+                  this.getProfiles(data.controlName || '').forEach(
```

Comment:
prefer for..of over foreach

https://2ality.com/2021/01/looping-over-arrays.html